www/js/app.js

```diff
@@ -386,6 +397,36 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
         }, 600);
     });
 
+    //focus search bar (#prefix) if Home key is pressed
+    function focusPrefixOnHomeKey(event) {
+        //check if home key is pressed
+        if (event.key === 'Home') {
+            //scroll to top of #search-article section if scroll bar is associated with it
+            $('#search-article').scrollTop(0);
+            //in case the scroll bar is associated with iframe #articleContent
+            var iframeArticleDocument = document.getElementById('articleContent').contentWindow.document;
+            $(iframeArticleDocument).scrollTop(0);
+            $('#prefix').focus();
+        }
+    }
+    //switch on/off the feature to use Home Key to focus search bar
+    function switchHomeKeyToFocusSearchBar() {
+        var iframeContentWindow = document.getElementById('articleContent').contentWindow;
+        // when the feature is in active state
+        if (params.useHomeKeyToFocusSearchBar) {
+            //Handle Home key press inside window(outside iframe) to focus #prefix
+            window.addEventListener('keydown', focusPrefixOnHomeKey);
+            //handle home key press inside iframe window
+            iframeContentWindow.addEventListener('keydown', focusPrefixOnHomeKey);
```

Comment:
I'm wondering if we should also turn it on/off here in the docBody.
I think the way our UI is currently setup makes it unnecessary (because the current page is reloaded after coming back from Settings section).
But I'd be in favor of adding it here, in case we change this UI behavior.
In this case, a comment should explain why we have to set the eventListener on both the iframeContentWindow and the docBody

Reply:
Sorry, for not writing an explanation for this code. It will be done.

It is not required to set event listener on both `iframeContentWindow` and `docBody`. 
This part was included to handle the initial iFrame which is displayed when no zim file is loaded yet.
For loading which none of the function `readArticle` or `displayArticleContentInIframe` are called. This is to be handled when document is ready and when the Home-Key-Feature is turned on/off.
Also, Since the Zim Contents are reloaded each time we visit the HomePage. The eventlistener associated with `iFrameContentWindow` will also be removed on the reload itself hence does not remain useful (given current UI behaviour) in any case other than initial iframe.